### PR TITLE
Fail silently when unable to find update.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

File: src/Command/CommandBase.php [hunk 1]
```diff
@@ -1192,22 +1192,26 @@ protected function checkForNewVersion(InputInterface $input, OutputInterface $ou
...
-    $response = $client->get('https://api.github.com/repos/acquia/cli/releases');
-    $releases = json_decode($response->getBody());
-
-    if (!isset($releases[0])) {
-      throw new \Exception('API error - no release found at GitHub repository acquia/cli');
+  protected function hasUpdate(): bool {
+    try {
+      $client = $this->getUpdateClient();
+      // Intentionally silence and allow errors. It's ok if this fails.
+      $response = @$client->get('https://api.github.com/repos/acquia/cli/releases');
```
Comment (anavarre) [inline src/Command/CommandBase.php:1201]: I tried changing this URL to a one that doesn't exist to test out this PR. Didn't work despite wiping `vendor`, then doing a `ckc` and `cc`. Any idea how to try running into this issue?

File: src/Command/CommandBase.php [hunk 2]
```diff
@@ -1175,11 +1175,11 @@ protected function checkForNewVersion(InputInterface $input, OutputInterface $ou
       return;
     }
     // Bale in Cloud IDEs to avoid hitting Github API rate limits.
-    if (!AcquiaDrupalEnvironmentDetector::isAhIdeEnv()) {
```
Comment (grasmash) [inline src/Command/CommandBase.php:1178]: @anavarre 🤔 Seems important to fix.

Comment (anavarre) [reply src/Command/CommandBase.php:1178]: Without this IDEs were constantly asking to update and customers were doing it, which had us run into API limits without a few hours of an ACLI release.

Comment (grasmash) [reply src/Command/CommandBase.php:1178]: I'm not removing this line, I'm removing the `!`

Comment (grasmash) [reply src/Command/CommandBase.php:1178]: When I acli `master` in an IDE, the code does (incorrectly) progress to execute `$this->hasUpdate()`.

File: src/Command/CommandBase.php [hunk 3]
```diff
@@ -1170,16 +1170,16 @@ protected function checkForNewVersion(InputInterface $input, OutputInterface $ou
...
     }
-    // Bale for development builds.
+    // Bail for development builds.
     if ($this->getApplication()->getVersion() == '@package_version@') {
       return;
     }
-    // Bale in Cloud IDEs to avoid hitting Github API rate limits.
-    if (!AcquiaDrupalEnvironmentDetector::isAhIdeEnv()) {
+    // Bail in Cloud IDEs to avoid hitting Github API rate limits.
+    if (AcquiaDrupalEnvironmentDetector::isAhIdeEnv()) {
```
Comment (danepowell) [inline src/Command/CommandBase.php:1178]: We don't have to solve this here, but I don't like that ACLI has hardcoded support for IDEs. I'd rather support an environment variable like `ACLI_DISABLE_UPDATES`, which could be set in any environment.

File: src/Command/CommandBase.php [hunk 4]
```diff
@@ -1170,16 +1170,16 @@ protected function checkForNewVersion(InputInterface $input, OutputInterface $ou
...
       return;
     }
     try {
-      if ($this->hasUpdate()) {
+      if (@$this->hasUpdate()) {
```
Comment (danepowell) [inline src/Command/CommandBase.php:1182]: I don't feel good about this error suppression. We already have so much trouble debugging failed update checks.

Is there a way to tie this to verbosity? So we can at least see errors in verbose output.

File: src/Command/CommandBase.php (hunk 3 above)
Comment (grasmash) [reply src/Command/CommandBase.php:1178]: ACLI has hardcoded support for IDEs in _a lot_ of places. Here we're just fixing a but, not integrating it more deeply.

File: src/Command/CommandBase.php (hunk 4 above)
Comment (grasmash) [reply src/Command/CommandBase.php:1182]: Not that I know of. The problem is that you can't catch some php error and warnings.

**Review by danepowell — APPROVED** [review]

**Pull request merged**